src/sentry/utils/performance_issues/base.py

```diff
@@ -167,19 +167,44 @@ def get_duration_between_spans(first_span: Span, second_span: Span):
 
 
 def get_url_from_span(span: Span) -> str:
+    """
+    Parses the span data and pulls out the URL. Accounts for different SDKs and
+    different versions of SDKs formatting and parsing the URL contents
+    differently.
+    """
+
     data = span.get("data") or {}
-    url = data.get("url") or ""
-    if not url:
-        # If data is missing, fall back to description
-        description = span.get("description") or ""
-        parts = description.split(" ", 1)
-        if len(parts) == 2:
-            url = parts[1]
-
-    if type(url) is dict:
-        url = url.get("pathname") or ""
-
-    return url
+
+    # The most modern version is to provide URL information in the span
+    # data
+    url_data = data.get("url")
+
+    if type(url_data) is dict:
+        # Some transactions mysteriously provide the URL as a dict that looks
+        # like JavaScript's URL object
+        url = url_data.get("pathname") or ""
+        url += url_data.get("search") or ""
+        return url
+
+    if type(url_data) is str:
+        # But usually the URL is a regular string, and so is the query. If
+        # `http.query` is absent, `url` contains the query
+        url = url_data
+        query_data = data.get("http.query")
```

Comment:
I'd maybe mention this is the preferred way of doing it moving forward.

Reply:
Makes sense, done